### ISSUE TITLE
throws is clunky to use

### ISSUE THREAD

**Issue by keis** [rluba/hamjest#14]
the throws matches in hamjest is a bit clunky to use compared to assertThrows of other frameworks.

```
   assertThat(foo, throws(allOf(instanceOf(Error),
                                instanceOf(MyCustomError),
                                hasProperty('message', matchesPattern(/but .%. found/))))
```
compared to
```
  assertThrows(foo, MyCustomError, /but .%. found/)
```

however I do like that is uses a submatcher as it gives a lot of flexibility. I wrote a small `error` matcher for my own use and that works ok but it would be nice if something similiar was supported out of the box.

https://gist.github.com/keis/11a3028eae3217214fc5

**Comment by rluba** [comment]
Thank you for the feedback! I'll seriously consider adding something similar to your `error` matcher to _hamjest_.

**Comment by rluba** [comment]
Its implemented in `v2.13.0` as `typedError(…)`.
